power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_data.hpp

```diff
@@ -51,18 +51,18 @@ template <class T> constexpr CType ctype_v = ctype_t<T>::value;
 // function selector based on ctype
 // the operator() of the functor should have a single template parameter
 // the selector will instantiate the operator() with relevant type
-template <class Functor, class... Args> decltype(auto) ctype_func_selector(CType ctype, Functor f, Args&&... args) {
+template <class Functor, class... Args> decltype(auto) ctype_func_selector(CType ctype, Functor&& f, Args&&... args) {
     using enum CType;
 
     switch (ctype) {
     case c_double:
-        return f.template operator()<double>(std::forward<Args>(args)...);
+        return std::forward<Functor>(f).template operator()<double>(std::forward<Args>(args)...);
     case c_double3:
-        return f.template operator()<RealValue<asymmetric_t>>(std::forward<Args>(args)...);
+        return std::forward<Functor>(f).template operator()<RealValue<asymmetric_t>>(std::forward<Args>(args)...);
     case c_int8:
-        return f.template operator()<int8_t>(std::forward<Args>(args)...);
+        return std::forward<Functor>(f).template operator()<int8_t>(std::forward<Args>(args)...);
     case c_int32:
-        return f.template operator()<int32_t>(std::forward<Args>(args)...);
+        return std::forward<Functor>(f).template operator()<int32_t>(std::forward<Args>(args)...);
```

Comment:
Was this refactor done here to align with the other implementations of `func_selector`s? Because this doesn't necessarily aligns with this scope, does it?

Reply:
> Was this refactor done here to align with the other implementations of func_selectors?

yes. sonar cloud was complaining for the other PR (because we didn't do perfect forwarding even though we could. I also updated this PR to ensure that now as well, it is perfectly forwarded. This comes with 2 benefits:

* performance benefits, because the whole function does not need to be copy-pasted every single time
* usability benefits, because lifetimes are preserved when using perfect forwarding